back/app/services/invites_service.rb

```diff
@@ -262,13 +262,15 @@ def build_invites(hash_array, default_params={}, inviter=nil)
       invites = hash_array.map do |invite_params|
         build_invite(invite_params, default_params, inviter)
       end
-      # Since invites will later be created in a single transaction, the
-      # normal mechanism for generating slugs could result in non-unique
-      # slugs. Therefore we generate the slugs manually
+      # Since invites will later be created in a single transaction, the normal mechanism for
+      # generating slugs could result in non-unique slugs. Therefore, we generate the slugs manually,
+      # unless Abbreviated User Names feature is enabled.
       invitees = invites.map(&:invitee)
-      invitees.zip(SlugService.new.generate_slugs(invitees){|u| u.full_name}) do |(invitee, slug)|
-        if invitee.full_name.present?
-          invitee.slug = slug
+      if AppConfiguration.instance.feature_activated?('abbreviated_user_names')
```

Comment:
With this check `if AppConfiguration.instance.feature_activated?('abbreviated_user_names')`, we introduce duplication with `User#generate_slug`.

Ideally, all this logic should be in one method or at least in one class.

Reply:
Although, now I look at it, i don't know why we weren't just calling `user#generate_slug` here anyway 🤔

Reply:
The problem is, as stated in the comments, that if we simply do...
```
invitees = invites.map(&:invitee)
invitees.each { |invitee| invitee.generate_slug }
invites
```
invitees = invites.map(&:invitee)
invitees.each { |invitee| invitee.generate_slug }
invites
```
We get duplicate slugs from duplicate names. I guess because we aren't saving each user after each slug value is assigned, hence uniqueness validation isn't applied.

Reply:
I believe all the logic related to slugs generation (anonymized or not) should be in `SlugService`.
1. It will prevent forgetting to use checks like `if AppConfiguration.instance.feature_activated?('abbreviated_user_names')`. Now they are scattered over the project in two different places.
2. It encapsulates all the logic in one place. So, when we need to find something related to slugs, we just go there.
3. It simplifies testing.

Reply:
OK. Makes sense. I will try that. I guess I will need to check for `type of object == User` in `SlugService`

Reply:
@alexander-cit I think I might disagree. I think it makes sense that `SlugService` takes in some strings and produces a slug (and also checks the slug is unique & adds suffix to make unique if needed.)

I don't think it makes sense to move logic of [pseudocode]
```
if a user record
  if abbreviated names enabled
    return uuid
  else if user.full_name?
    generate slug from user.full_name
  rnd
else
  regular slug generation flow
end
```
To a generic module like `SlugService`

Reply:
For example, the `generate_slugs` method in `SlugService` is called by the `invites_service` `build_invutes` method we are considering, and it expects a block that specifies the field to use to build the slug along with the collection provided. How do we both deal with the anonymized case, where no block is needed as we don't need the `full_names` to create the slugs, and the regular case where we do need the block?

This is just one complicating factor in the refactoring you propose.